### PR TITLE
Helper function for dropping timeslice

### LINKED ISSUE #363: Timeslice level names are no longer configurable [BUG]
## Describe the bug

Previously you could choose the name of the timeslice levels using the `level_names` argument in `settings.toml`. However, since #289, the default level names ("month", "day" and "hour") are hardcoded in many different places in the code, but we still accept the `level_names` argument in `settings.toml`. If anyone tries to change this, the model will fail.

I think before it would also accept fewer or more timeslice levels (at least according to the documentation, although I haven't tried this), but obviously now it's restricted to three levels.

## To Reproduce

Try changing `level_names` for any of the example models

## Expected behavior

We should either change the code to stop hardcoding level names, or remove the `level_names` option from `settings.toml` so the defaults ("month", "day" and "hour") are mandated.

I'd probably go with the latter, as it doesn't seem like there would be much use for changing the level names, but maybe I'm wrong?

## Context

Please, complete the following to better understand the system you are using to run MUSE.

- Operating system (eg. Windows 10):
- MUSE version (eg. 1.0.1):
- Installation method (eg. pipx, pip, development mode):
- Python version (you can get this running `python --version`):

### PR DESCRIPTION
**Pull request by tsmbland**
# Description

Adds a helper function for dropping the timeslice variable from data arrays, and uses this throughout the code instead of `drop_vars(["timeslice", "month", "day", "hour"])`

This partially fixes #363, but a few other changes will be required which I'm addressing in #381

## Type of change

Please add a line in the relevant section of
[CHANGELOG.md](https://github.com/EnergySystemsModellingLab/MUSE_OS/blob/development/CHANGELOG.md) to
document the change (include PR #) - note reverse order of PR #s.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (non-breaking, back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (whatever its nature)

## Key checklist

- [ ] All tests pass: `$ python -m pytest`
- [ ] The documentation builds and looks OK: `$ python -m sphinx -b html docs docs/build`

## Further checks

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: src/muse/timeslices.py [hunk 1]
```diff
@@ -596,4 +596,15 @@ def represent_hours(
     return convert_timeslice(DataArray([nhours]), timeslices).squeeze()
 
 
+def drop_timeslice(data: DataArray):
```
Comment (dalonsoa) [inline src/muse/timeslices.py:599]: Ideally, add output type:
```suggestion
def drop_timeslice(data: DataArray) -> DataArray:
```

**Review by dalonsoa — CHANGES_REQUESTED** [review]
Given that you're on it, add a dedicated tests that checks the new function.

File: src/muse/timeslices.py [hunk 2]
```diff
@@ -596,4 +596,15 @@ def represent_hours(
...
+    """Drop the timeslice variable from a DataArray.
+
+    If the array doesn't contain the timeslice variable, return the input unchanged.
+    """
+    if "timeslice" not in data.dims:
+        return data
+
+    return data.drop_vars([c for c in data.timeslice.indexes])
```
Comment (alexdewar) [inline src/muse/timeslices.py:607]: Would this work:

```suggestion
    return data.drop_vars(data.timeslice.indexes)
```

If not, how about:

```suggestion
    return data.drop_vars(list(data.timeslice.indexes))
```

**Review by alexdewar — APPROVED** [review]
I've made a small suggestion (that probs won't work) but other than that, LGTM.

**Comment by tsmbland** [comment]
@dalonsoa Would you mind taking a quick look at this so I can merge it?

**Review by dalonsoa — APPROVED** [review]
LGTM! Sorry, I didn't realized the changes had been implemented.

**Pull request merged**